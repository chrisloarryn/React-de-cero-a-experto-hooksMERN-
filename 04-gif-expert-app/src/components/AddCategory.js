import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('Hello World')

  const handleInputChange = e => {
    e.preventDefault()
    setInputValue(e.target.value)
    // console.log('handleInputChange invoked')
  }

  const handleSubmit = e => {
    e.preventDefault()
    // console.log('handleSubmit', inputValue)
    setCategories(cats => [{ name: inputValue }, ...cats])
    setInputValue('')
  }

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input type='text' value={inputValue} onChange={handleInputChange} />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
