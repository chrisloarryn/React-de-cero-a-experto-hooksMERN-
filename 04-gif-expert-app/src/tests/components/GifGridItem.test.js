import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import { GifGridItem } from './../../components/GifGridItem'

describe('Test into <GifGridItem /> component', () => {
  const { title, url } = { title: 'Hello', url: 'https://google.cl' }

  const wrapper = shallow(<GifGridItem title={title} url={url} />)
  // beforeEach(() => {
  //     wrapper = shallow(<GifGridItem />)
  // })

  test('test into the component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should return paragraph with the title', () => {
    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(title)
  })

  test('should get img = to alt and url from props', () => {
    const img = wrapper.find('img')
    // console.log(img.props()) // prop('src') => for only one prop
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })

  test(`should have 'animate__delay-1s' className`, () => {
      const div = wrapper.find('div')
      const className = div.prop('className')
      expect(className.includes('animate__delay-1s')).toBe(true)
      expect(className.includes('animate__delay-1s')).not.toBe(false)
  })
})
