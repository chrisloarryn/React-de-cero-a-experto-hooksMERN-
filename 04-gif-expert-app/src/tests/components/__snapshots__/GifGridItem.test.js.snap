// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test into <GifGridItem /> component test into the component 1`] = `
<div
  className="card animate__animated animate__bounceIn animate__delay-1s"
>
  <img
    alt="Hello"
    src="https://google.cl"
  />
  <p>
    Hello
  </p>
</div>
`;
